conrod_core/src/widget/primitive/shape/polygon.rs

```diff
@@ -232,6 +232,14 @@ impl<I> Widget for Polygon<I>
                     state.update(|state| state.points.truncate(total)),
                 None => (),
```

Comment:
Can we add an early return to this `None` branch so that we avoid doing the following `state.update` calls in the case that there has been no change to the widget instance? Calls to `state.update` indicate to conrod that the widget has changed in some way and requires re-drawing, so it's good to minimize these calls whenever possible.